src/plugins/home/public/services/section_type/section_type.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CoreStart } from 'opensearch-dashboards/public';
+import { BehaviorSubject, Observable, Subscription, combineLatest } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { EuiLinkAnchorProps } from '@elastic/eui';
+import { DataPublicPluginStart } from '../../../../data/public';
+import { SavedObjectLoader } from '../../../../saved_objects/public';
+import { createSavedHomepageLoader, SavedHomepage } from '../../saved_homepage';
+
+// TODO: this should support error handling explicitly
+// TODO: this should support async rendering through a promise
+export type RenderFn = (element: HTMLElement) => () => void;
+
+export interface HeroSection {
+  id: string;
+  render: RenderFn;
+}
+
+export interface Section {
+  id: string;
+  title: string;
+  description?: string;
+  links?: Array<{ label: string; url: string; props?: Omit<EuiLinkAnchorProps, 'href'> }>;
+  render: RenderFn;
+}
+
+export interface Homepage {
+  heroes$: Observable<HeroSection[] | undefined>;
```

Comment:
For the hero, why would happen if multiple different plugin register hero for themselves, which hero should we display? In my understanding, we should just display one hero per screen?

Reply:
I dont think that was a requirement. Maybe based on how the homepage evolves we may restrict that. cc @kgcreative ?